constrained-voucher.mkd

```diff
@@ -520,6 +506,92 @@ One key difference is that the names of the leaves in the YANG does not have a m
+Note that these are two goals, and these two goals may be at odds.
+
+One piece of code that is complex, hard to test, and never seems critical for application design is the PKIX certification validation code.
+This section explains how a device can be onboarded with the minimal amount of PKIX related code.
+
+## The IDevID artifact
+
+The BRSKI protocol is based upon the Pledge identifying itself with an RFC5280 format (X509v3, PKIX) certificate.
+The details of how it becomes an IDevID is defined in {{ieee802-1AR}}.
+While the Pledge device needs to have an IDevID object, it does not need to manipulate it.
+It can regard the IDevID as an opaque object which is placed into a DTLS setup.
+
+The IDevID certificate will have an associated private key.
+The Pledge will need to use that private key for the DTLS connection, and for signing of the voucher-request.
+However, none of these operations require any kind of manipulation of certificates.
+The private key can be loaded in whatever format is appropriate and efficient for the hardware.
+(EDNOTE: are there lwig references that would make sense here?)
+
+## The Registrar Trust Anchor
+
+When the Pledge first connects to the Registrar, the connection to the Registrar is provisional, as explained in section 5.6.2 of {{I-D.ietf-anima-bootstrapping-keyinfra}}.
+The Registrar provides its public key in a TLSServerCertificate, and the Pledge uses that to validate that integrity of the (D)TLS connection, but it does not validate the identity of
+the provided certificate.
+
+If instead of using a (TLSServer)Certificate of type X509 (see section 4.4.2 of {{RFC8446}}),
+a RawPublicKey object is used, then the Pledge does not need to any decoding of the RF5280 format certifiate to get the public key.
+
+## The Voucher Request
+
+The Pledge can put the Registrar's public key into the proximity-registrar-subject-public-key-info
+(PRSPKI) field of the voucher-request.
+(The proximity-registrar-sha256-of-subject-public-key-info can also be used if the 32-bytes of hash turned out to be smaller than a typical ECDSA key.)
+
+As the format of the PRSPKI field is identical to the TLS Certificate RawPublicKey, no manipulation at all is needed to insert this into a voucher-request.
+
+## The Voucher Reply
+
+A returned voucher will have a pinned-domain-subject-public-key-info field with the identical key as was found in the proximity field above, as well as in the TLS connection.
+Validation of this is what takes the connection out of the provisional state (see {{I-D.ietf-anima-bootstrapping-keyinfra}} section 5.6.2).
+
+The voucher needs to be validated first.
+The Pledge needs to have a public key to validate the signature from the MASA on the voucher.
+
+Certificate chains may be included in the COSE-signed Constrained Vouchers described in this document, using the x5bag attribute from {{I-D.ietf-cose-x509}}.
```

Comment:
If possible we should leave this out! We only define, AFAIK, using the x5bag for the Registrar to send a bag of certs (including one chain) to MASA in a Voucher Request. So not for the BRSKI-EST protocol. And not in Vouchers.
To avoid confusion, let's remove - we are now in the RPK section so no need to say how the non-RPK protocol looks like.

Reply:
Update: I checked RFC 8366 and indeed the CMS structure there may include multiple certificates as needed for the Pledge, to build a chain of trust between the MASA's signing identity used and the pre-installed TA in the Pledge. Additionally the TA as used by the Pledge should be included in the CMS structure. 

In the constrained case, using COSE signing, typically no cert is included in the COSE structure but just the signature and (if really needed) a key identifier.  So to me it is still not clear why we would want to put any certificates in a Voucher in an x5bag at all ; given that we are now in the RPK section and thus one of the below conditions hold:

1) Pledge requires RPK-pinning and is unable to perform X.509 chaining
2) RPK-pinning is used to reduce size of Voucher  (putting certs in x5bag isn't going to reduce ... )
3) Pledge authenticates using only its RPK, Registrar does this too per RFC 7250, Registrar's RPK is pinned.

In conclusion I still don't see how this text on x5bag and cert chains would fit here.